### PR TITLE
DOC: Copy-edit the 'partition' docstring.

### LINKED ISSUE #7958: Slight inaccuracy in the `partition` docstring.
The docstring for the `partition` method says:

```
All elements smaller than the kth element are moved before this element and
all equal or greater are moved behind it.
```

That's not quite correct, because some of the elements before the `kth` element could be equal to the `kth` element.  A trivial example is an input containing all the same values.  Perhaps a better wording is something like:

```
In the partitioned array, all elements before the `kth` element are less than or
equal to that element, and all the elements after the `kth` element are greater than
or equal to that element.
```

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Closes gh-7958.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
Link to the rendered updated docstring: [partition](https://output.circle-artifacts.com/output/job/76a3a7cb-2914-4483-a234-c6da2e65bf78/artifacts/0/doc/build/html/reference/generated/numpy.partition.html)

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -679,10 +679,11 @@ def partition(a, kth, axis=-1, kind='introselect', order=None):
 
     Creates a copy of the array with its elements rearranged in such a
     way that the value of the element in k-th position is in the
-    position it would be in a sorted array. All elements smaller than
-    the k-th element are moved before this element and all equal or
-    greater are moved behind it. The ordering of the elements in the two
-    partitions is undefined.
+    position it would be in a sorted array.  In the partitioned array, all
```
Comment (charris) [inline numpy/core/fromnumeric.py:682]: "it" is a bit ambiguous, the element or its value.  I think "value" would be correct, i.e., the same as the value of the k-th element of the sorted array.

Comment (WarrenWeckesser) [reply numpy/core/fromnumeric.py]: Thanks @charris, I pushed an update.  I also fixed `kth` to `k-th` to match the existing usage.

**Comment by charris** [comment]
Thanks Warren.

**Pull request merged**